### PR TITLE
Adjust media upload source for rich text

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

1) Fixes media blocks to use `rich-text` source on upload.
2) Fixes file block transforms.
3) Updates some docs to use `rich-text` instead of `children`.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

File: packages/block-library/src/file/index.js [hunk 1]
```diff
@@ -109,7 +109,7 @@ export const settings = {
 				transform: ( attributes ) => {
 					return createBlock( 'core/file', {
 						href: attributes.src,
-						fileName: attributes.caption && attributes.caption.join(),
+						fileName: attributes.caption,
```
Comment (iseulde) [inline packages/block-library/src/file/index.js]: Note that this wouldn't even work for more complex `rich-text` values...

File: packages/block-library/src/image/index.js [hunk 2]
```diff
@@ -158,7 +158,7 @@ export const settings = {
 						selector: 'img',
 					},
 					caption: {
-						type: 'array',
+						type: 'rich-text',
 						// To do: needs to support HTML.
```
Comment (youknowriad) [inline packages/block-library/src/image/index.js]: Should we remove this comment now?

Comment (iseulde) [reply packages/block-library/src/image/index.js]: Why is it wrong? If you have an attribute source you can have different types as well, that's why you can declare a different type?

File: packages/block-library/src/file/index.js [hunk 3]
```diff
@@ -143,7 +143,7 @@ export const settings = {
 				transform: ( attributes ) => {
 					return createBlock( 'core/audio', {
 						src: attributes.href,
-						caption: [ attributes.fileName ],
+						caption: attributes.fileName,
```
[the remark this answers is not in the document]
Comment (iseulde) [reply packages/block-library/src/file/index.js]: No, they are both rich text values. It's strange that the file name is not a plain text input field, but that's not in the scope of this PR.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by jorgefilipecosta** [comment]
As soon as an image is uploaded in the image/gallery placeholder, using the upload button we get a crash. I think we may need to apply create function from the RichText package when setting the attributes after the upload like we did in for the FileBlock.

**Comment by iseulde** [comment]
@jorgefilipecosta I updated and rebased the PR with your latest changes.

**Review by jorgefilipecosta — APPROVED** [review]
Thank you for iterating on this changes @iseulde. Everything looks good in my tests now 👍

**Comment by iseulde** [comment]
Thanks @jorgefilipecosta! :)

**Pull request merged**